### PR TITLE
Add resolved search type response field

### PR DESCRIPTION
**Pull request by jeffzwang**

### CODE REVIEW

File: exa_py/api.py [hunk 1]
```diff
@@ -394,15 +399,20 @@ class SearchResponse(Generic[T]):
     Attributes:
         results (List[Result]): A list of search results.
         autoprompt_string (str, optional): The Exa query created by the autoprompt functionality.
+        resolved_search_type (str, optional): What "auto" saerch resolved to. "neural" or "keyword".
```
Comment (mfine15) [inline exa_py/api.py]: typo in search

File: setup.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 setup(
     name="exa_py",
-    version="1.0.16",
+    version="1.0.18",
```
Comment (mfine15) [inline setup.py:5]: why the double version bump

File: exa_py/api.py [hunk 3]
```diff
@@ -41,18 +41,21 @@ def snake_to_camel(snake_str: str) -> str:
...
 
     Args:
         data (dict): The dictionary with keys in snake_case format.
 
     Returns:
         dict: The dictionary with keys converted to camelCase format.
     """
-    return {snake_to_camel(k): v for k, v in data.items() if v is not None}
-
+    return {
```
Comment (mfine15) [inline exa_py/api.py:54]: is this just black auto format? shouldn't the whole repo already be like that

File: exa_py/api.py [hunk 4]
```diff
@@ -430,7 +440,7 @@ def __init__(
         self,
         api_key: Optional[str],
         base_url: str = "https://api.exa.ai",
-        user_agent: str = "exa-py 1.0.16",
+        user_agent: str = "exa-py 1.0.18",
```
Comment (mfine15) [inline exa_py/api.py:443]: should probably just read this from setup.py/file so that it's always in sync

**Comment by mfine15** [comment]
lgtm after the search typo fixed

**Review by mfine15 — APPROVED** [review]

File: setup.py (hunk 2 above)
Comment (jeffzwang) [reply setup.py:5]: prev was not merged to master i think rip

File: exa_py/api.py (hunk 4 above)
Comment (jeffzwang) [reply exa_py/api.py:443]: ok lemme just ship this fast, yes that would be nice

**Pull request merged**